<center><h1>Statistic</h1></center>

<img src="https://fiverr-res.cloudinary.com/images/t_main1,q_auto,f_auto/gigs/105100081/original/b508ab6dd88851cf41a84b367f661443a02aa202/provide-you-best-solution-about-stat-math-and-computer.jpg"/>

## Central Limit Theorem

## Point Estimate

<center><h1>Confidence Interval</h1></center>

### Case- 1:- Variance is known

### Example:-
A sample of 11 circuits from a normal population has a mean resistance of 2.20 ohms.  We know from past testing that the population standard deviation is 0.35 ohms.  Determine a 95% confidence interval for the true mean resistance of the population.


Solution:-these example is under known variance.
<img src="http://knowledgehills.com/content/images/Six-Sigma/Six-Sigma-Z-Confidence-Interval-Means-Formula1.gif"/>


<img src="https://www.statisticshowto.datasciencecentral.com/wp-content/uploads/2014/12/ci95.png"/>

In [79]:
#Mean Interval Estimation using Normal Distribution
import math
size_of_sample=11
standard_deviation=0.35
poe=2.20
standard_error=standard_deviation/math.sqrt(size_of_sample)
critical_value=scipy.stats.norm.ppf(0.025)
moe=critical_value*standard_error
CI=(poe+moe,poe-moe)
print("Confidence Interval:-",CI)



Confidence Interval:- (1.9931670182940682, 2.406832981705932)


In [80]:
#Mean Interval Estimation using Normal Distribution
import statsmodels.stats.api as sms
x = range(10,50)
ci=scipy.stats.norm.interval(0.95, loc = np.mean(x), scale = scipy.stats.sem(x))
lower, upper = sms.DescrStatsW(x).zconfint_mean(alpha = 0.05)
print("Confidence Interval:-",ci)
print("lower:",lower,"\nUpper:",upper)

Confidence Interval:- (25.877157979701813, 33.12284202029819)
lower: 25.87715797970181 
Upper: 33.12284202029819


### Case- 2:- Variance is Unknown

<img src="http://www.six-sigma-material.com/images/CIMeanExampleCalculation.JPG"/>

In [81]:

#Mean Interval Estimation using t dist
# add most useful things degree of freedome=n-1
import numpy as np
import scipy.stats
import statsmodels.stats.api as sms
x = range(10,50)
scipy.stats.t.interval(0.95, len(x) - 1, loc = np.mean(x), scale = scipy.stats.sem(x))
Lower, Upper = sms.DescrStatsW(x).tconfint_mean(alpha = 0.05)
print("lower:",Lower,"\nUpper:",Upper)

lower: 25.761212100370575 
Upper: 33.23878789962942


<center><h1>Hypothesis Testing</h1></center>

A (statistical) hypothesis is a statement or claim about a population parameter (not
about a sample statistic):<br>
Ex) The mean electric bill per household of this city is μ = $132.
The Mean weight of chocolate bars manufactured in a factory is 3 ounces.

#### Hypothesis testing
    A procedure leading to a decision about a hypothesis based on a random sample

#### Null  Hypothesis  
(H0)  states the assumption to be tested.	A hypothesis testing begins  with the assumption that H0 is true

#### Alternative Hypothesis 
(H1) is the opposite of the null hypothesis. It is the hypothesis  that the researcher is trying to prove.

Two-sided Alternative Hypothesis


### Case- 1`:- Variance is known(Using Z -Distribution)

### Example (using Critical value & P-value)
To test the claim that the mean weight of chocolate bars manufactured in a  factory is 3 ounces, we weighed 100 chocolate bars and the average weight was 2.84.  Suppose that, from past records, the standard deviation is known to be 0.8.


### 1.State the null and alternative hypotheses
H0: μ = 3	H1: μ ≠ 3	(two-sided test)


### 2.Choose the desired level of significance
Suppose that  =0 .05 is chosen for this test


### 3.Determine the test statistic
σ is known so this is a Z-test
<img src="https://image.slidesharecdn.com/z-test-170412150226/95/ztest-with-examples-4-638.jpg?cb=1546230216"/>

In [83]:
Z = (2.84 - 3) / (0.8/math.sqrt(100))
print("Test_STatistic",Z)

Test_STatistic -2.0000000000000018


### 4.Find the critical values and determine the rejection region(s)
For  = 0.05 , the critical alpha/2=0.25

In [84]:
scipy.stats.norm.ppf(0.025)
# or both will use only difference in -ve and +ve sign
scipy.stats.norm.ppf(0.975)

1.959963984540054

Reject H0 if z0 < -1.96 or z0 > 1.96

### 5.Reach a decision and interpret the result
Since z0 = -2.0 < -1.96, you reject the null hypothesis
That is, there is sufficient evidence that the mean weight of chocolate bars is not equal to 3.

### Aproach-ii Using P-Value

NOTE:- <br>
P-Value > aplha----acept.<br>
P-Value > alpha----reject.

In [85]:
# p-value = 2P(Z > lz0l ) =2P(Z>2.0) because its 2 sided so multply by 2
P_Value=2*scipy.stats.norm.cdf(-2.0)
print("P value:-",P_Value)

P value:- 0.04550026389635839


In [86]:
# now compare with alpha and P value
# P-Value > aplha----acept.
# P-Value > alpha----reject.
alpha=0.05
P_Value=0.04550026389635839
if(P_Value < alpha):
    print("Thus, we reject the null hypothesis\nThat is, there is sufficient evidence that the mean weight of chocolate bars is not equal to 3")
else:
    print("Accept Null hypothesis")

Thus, we reject the null hypothesis
That is, there is sufficient evidence that the mean weight of chocolate bars is not equal to 3


### Case- 2`:- Variance is Unknown(Used T-Distribution)

### Example 
The mean cost of a hotel room in LA is said to be $168 per night. A random  sample of 25 hotels resulted in X = 172.50 and S = 15.40. Test at the = 0.05 level  Assuming the data are normally distributed.


### 1.State the null and alternative hypotheses
H0: μ = 168	H1: μ != 168


### 2.Choose the desired level of significance
Suppose that  =0 .05 is chosen for this test


### 3.Determine the test statistic
σ is Unknown so this is a T-test
<img src="https://image1.slideserve.com/1517587/the-formula-for-the-test-statistic-with-n.jpg"/>

In [87]:
import math
Mu=168
n=25
s=15.04
alpha=0.05
xbar=172.50

T=(172.50-168)/((15.40)/math.sqrt(25))
print("T Statistic",T)

T Statistic 1.461038961038961


### 4.Find the critical values and determine the rejection region(s)
For  = 0.05 , the critical alpha/2=0.25

In [88]:
# degree of freedome=n-1
dof=25-1
alpha=0.05 
al=0.05/2   # because its two sided
# scipy.stats.t.ppf(alpha,degree of freedom)
critical_Value=stats.t.ppf(0.975,dof)
print("Critical Value:-",critical_Value)

Critical Value:- 2.0638985616280205


Critical Values: t0.025, 24 = ± 2.0639<br>
Reject H0 if	t0>2.0639 or t0<-2.0639


### 5.Reach a decision and interpret the result
Since t0= 1.4960106382978724 < 2.0638985616280205,<br> 
Since t0 does not fall in the rejection region, we cannot reject H0.<br>
That is, there is sufficient evidence

### Aproach-2 Using P-Value

NOTE:- <br>
P-Value > aplha----acept.<br>
P-Value > alpha----reject.

In [89]:
P_V=1-stats.t.cdf(1.461038961038961,24)
P_Value=P_V*2  # because its two sides
print("P-Values:",P_Value)

P-Values: 0.1569743731561739


In [114]:
# now compare with alpha and P value
# P-Value > aplha----acept.
# P-Value > alpha----reject.
alpha=0.05
P_Value=0.1569743731561739
if(P_Value > alpha):
    print("Thus, we Accept the null hypothesis.Because P-Value gratter than alpha\nThat is, there is sufficient evidence.")
else:
    print("Reject Null hypothesis")

Thus, we Accept the null hypothesis.Because P-Value gratter than alpha
That is, there is sufficient evidence.


<center><h2>CHi -Square</h2></center>


Interference on the variance of a Normal Population.A standard normal deviate is a random sample from the standard normal distribution. The Chi Square distribution is the distribution of the sum of squared standard normal deviates. The degrees of freedom of the distribution is equal to the number of standard normal deviates being summed.<br>
H0: s2 = s02<br>
H1: s2 != s02<br>
where s02  is a specified constant.<br>
Sampling from a normal distribution with unknown mean m and unknown variance s2, the quantity



<img src="https://www.empirical-methods.hslu.ch/files/2017/02/calculating-the-chi-square-test-statistic-for-variance.png"/>

#### Example:
 An automatic filling machine is used to fill bottles with liquid detergent.  A random sample of 20 bottles results in a sample variance of fill volume of s2 = 0.0153 (fluid ounces)2.  If the variance of fill volume exceeds 0.01 (fluid ounces)2, an unacceptable proportion of bottles will be underfilled and overfilled.  Is there evidence in the sample data to suggest that the manufacturer has a problem with underfilled and overfilled bottles?  Use a = 0.05, and assume that fill volume has a normal distribution.

### 1: parameter of interest----> population of Variance

### 2.State the null and alternative hypotheses
H0:s2=0.01<br>
H1:s2>0.01 
Note:- this is the one sided.

### 3.Determine the test statistic
σ is Unknown so this is a T-test
<img src="https://www.empirical-methods.hslu.ch/files/2017/02/calculating-the-chi-square-test-statistic-for-variance.png"/>

In [111]:
alpha=0.05
n=20
sample=0.0153
sigma=0.01
x2=(19*sample)/sigma
print("Test statistic:",x2)

Test statistic: 29.07


### 4.Find the critical values and determine the rejection region(s)
For  = 0.05

In [110]:
# in calculation of critical value in chi sqaure his taking two parameter.
# scipy.stats.chi2.ppf(test statistics,degree of freedome)
scipy.stats.chi2.ppf(0.95,19)

30.14352720564616

### 5.Reach a decision and interpret the result
Because x2=29.07<x2(0.0153)=30.14<br> 
Accept the Null Hypothesis(H0)
Since t0 does not fall in the rejection region, we cannot reject H0.<br>
That is, there is sufficient evidence

### Aproach-2 Using P-Value

NOTE:- <br>
P-Value > aplha----acept.<br>
P-Value > alpha----reject.

In [115]:
P_Value=scipy.stats.chi2.cdf(29.07,19)
print("P_Value:",P_Value)

P_Value: 0.9351079950606133


In [116]:
# now compare with alpha and P value
# P-Value > aplha----acept.
# P-Value > alpha----reject.
alpha=0.05
P_Value=0.9351079950606133
if(P_Value > alpha):
    print("Thus, we Accept the null hypothesis.Because P-Value gratter than alpha\nThat is, there is sufficient evidence.")
else:
    print("Reject Null hypothesis")

Thus, we Accept the null hypothesis.Because P-Value gratter than alpha
That is, there is sufficient evidence.


<center><h1>Hypothesis Testing On Two Population</h1></center>

#### Example:
A high school math teacher claims that students in her class will score higher on the math portion of the ACT then students in a colleague's math class. The mean ACT math score for 49 students in her class is 22.1 and the standard deviation is 4.8. The mean ACT math score for 44 of the colleague's students is 19.8 and the standard deviation is t 5.4. At alpha = 0.10, can the teacher's claim be supported? Conduct the 5 steps hypothesis test and state your conclusion As an aid in the establishment of personnel requirements, the director of a hospital wishes to estimate the mean number of people who are admitted to the emergency room during a 24-hour period. The director randomly selects 24 different 24-hour periods and determines the number of admissions for each. For this sample, the mean is 19.8. The standard deviation is known to be 5. Find a 95% confidence interval for estimating the mean number of admissions per 24-hour period. Assume that systolic blood pressure (SBP) for healthy adults is normally distributed with a mean of 120 mm Hg and standard deviation of 20 mm Hg. What sample size is needed so that 95% of sample means are between 116 mmHg and 124 mmHg?<br>
x1bar=22.1  ,alpha=0.10<br>
x2bar=19.08 ,<br>
n1=49  ,s1=4.8  ,<br>
n2=44  ,s2=5.4  ,

### 1.State the null and alternative hypotheses
H0:Mu1<=mu2<br>
H1:mu1>mu2 <br>
Note:- this is the one sided.<br>
NOTE:- in this case we assuming that mu1<=mu1  so mu1-mu2=0

### 3.Determine the Standard Error
<img src="https://i.stack.imgur.com/W5GDJ.png"/>

In [121]:
SE=math.sqrt((4.8)**2/49+(5.4)**2/44)
print("Standard Error",SE)

Standard Error 1.0643924813525911


### 3.Determine the test statistic
σ is known so this is a Z-test
<img src="http://www.stat.yale.edu/Courses/1997-98/101/zstat2.gif"/>

In [124]:
Z=((22.1-19.8)-0)/SE
print("The Standardized Test Statistic:",Z)

The Standardized Test Statistic: 2.1608570525387822


### 4.Find the critical values and determine the rejection region(s)
For  = 0.10

In [126]:
Critical_Value=scipy.stats.norm.ppf(0.90)
print("Critical Value",Critical_Value)

Critical Value 1.2815515655446004


### 5.Reach a decision and interpret the result
The critical value =1.28 and Z0=2.162 so zo in rejection region<br>
Reject the Null Hypothesis(H0) and  Alternative Hypothesis<br>
Since Z0 does fall in the rejection region, we cannot accept Null Hypothesis(H0).<br>
That is, there is sufficient evidence